src/comps/rf.cc

```diff
@@ -0,0 +1,40 @@
+#include <lapack.hh>
+#include <RandBLAS.hh>
+#include <RandLAPACK.hh>
+
+#define USE_QR
+#define COND_CHECK
+#define VERBOSE
+
+namespace RandLAPACK::comps::rf {
+
+template <typename T>
+void RF<T>::rf1(
+    int64_t m,
+    int64_t n,
+    const std::vector<T>& A,
+    int64_t k,
+    std::vector<T>& Q
+){
+    using namespace blas;
+    using namespace lapack;
+
+    T* Omega_dat = RandLAPACK::comps::util::resize(n * k, this->Omega);
+    T* Q_dat = Q.data();
+
+    this->RS_Obj.call(m, n, A, k, this->Omega);
+
+    // Q = orth(A * Omega)
+    gemm<T>(Layout::ColMajor, Op::NoTrans, Op::NoTrans, m, k, n, 1.0, A.data(), m, Omega_dat, n, 0.0, Q_dat, m);
+
+    if(this->cond_check)
+        RandLAPACK::comps::util::cond_num_check(m, k, Q, this->Q_cpy, this->s, this->cond_nums, this->verbosity);
+    
+    // If CholQR failed, will use HQR
+    if(this->Orth_Obj.call(m, k, Q))
+        this->Orth_Obj.call(m, k, Q);
```

Comment:
Same comment as before: this assumes semantics of ``Orth_Obj`` where a second call uses a different algorithm if the first call returns a nonzero number. I think that short of thing is better handled internally by ``Orth_Obj``.